From b2a536c74719e3a7215838a7276991cafe1b421e Mon Sep 17 00:00:00 2001
From: Gunar Schorcht <gunar@schorcht.net>
Date: Thu, 8 Oct 2026 18:04:30 +0200
Subject: [PATCH 43/51] log: avoid redefinition of LOG_FORMAT macros

RIOT defines its own versions of LOG_FORMAT and LOG_SYSTEM_TIME_FORMAT.
Define the ESP-IDF versions only if they are not already defined.
---
 components/log/include/esp_log_format.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/components/log/include/esp_log_format.h b/components/log/include/esp_log_format.h
index 82e18139bb..92069ac99e 100644
--- a/components/log/include/esp_log_format.h
+++ b/components/log/include/esp_log_format.h
@@ -12,9 +12,13 @@ extern "C" {
 
 /** @cond */
 // For backward compatibility (these macros are not used in the log v2).
+#ifndef LOG_FORMAT
 #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
+#endif
 #define _ESP_LOG_DRAM_LOG_FORMAT(letter, format)  DRAM_STR(#letter " %s: " format "\n")
+#ifndef LOG_SYSTEM_TIME_FORMAT
 #define LOG_SYSTEM_TIME_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%s) %s: " format LOG_RESET_COLOR "\n"
+#endif
 /** @endcond */
 
 #ifdef __cplusplus
-- 
2.43.0

